src/main/java/org/example/stellog/review/service/ReviewService.java

```diff
@@ -0,0 +1,172 @@
+    private final ReviewMemberRepository reviewMemberRepository;
+
+    @Transactional
+    public void createReview(String email, Long roomId, ReviewRequestDto reviewRequestDto) {
+        Member member = memberRoomValidator.findMemberByEmail(email);
+        Room room = memberRoomValidator.findRoomById(roomId);
+        memberRoomValidator.validateMemberInRoom(member, room);
+        Starbucks starbucks = findStarbucksById(reviewRequestDto.starbucksId());
+
+        Review review = Review.builder()
+                .title(reviewRequestDto.title())
+                .content(reviewRequestDto.content())
+                .room(room)
+                .build();
+
+        ReviewMember reviewMember = ReviewMember.builder()
+                .member(member)
+                .review(review)
+                .build();
+
+        StarbucksReview starbucksReview = StarbucksReview.builder()
+                .review(review)
+                .starbucks(starbucks)
+                .build();
+
+        reviewRepository.save(review);
+        reviewMemberRepository.save(reviewMember);
+        starbucksReviewRepository.save(starbucksReview);
+    }
+
+    public ReviewListResponseDto getAllReviewsByStarbucksId(Long starbucksId) {
+        Starbucks starbucks = findStarbucksById(starbucksId);
+
+        List<StarbucksReview> starbucksReviews = starbucksReviewRepository.findAllByStarbucks(starbucks);
+
+        List<ReviewResponseDto> reviewResponseDtos = starbucksReviews.stream()
+                .map(StarbucksReview::getReview)
+                .map(review -> ReviewResponseDto.builder()
+                        .reviewId(review.getId())
+                        .starbucksId(starbucksId)
+                        .title(review.getTitle())
+                        .content(review.getContent())
+                        .build())
+                .toList();
+
+        return new ReviewListResponseDto(reviewResponseDtos);
+    }
+
+    public ReviewListResponseDto getAllReviewsByRoomId(String email, Long roomId) {
+        Member member = memberRoomValidator.findMemberByEmail(email);
+        Room room = memberRoomValidator.findRoomById(roomId);
+        List<Review> reviews = reviewRepository.findAllByRoom(room);
+        memberRoomValidator.validateMemberInRoom(member, room);
+
+        List<ReviewResponseDto> reviewResponseDtos = reviews.stream()
+                .map(review -> {
+                    StarbucksReview starbucksReview = starbucksReviewRepository.findByReview(review)
+                            .orElseThrow(StarbucksReviewNotFoundException::new);
+
+                    Long starbucksId = starbucksReview.getStarbucks().getId();
+
+                    return ReviewResponseDto.builder()
+                            .reviewId(review.getId())
+                            .starbucksId(starbucksId)
+                            .title(review.getTitle())
+                            .content(review.getContent())
+                            .build();
```

Comment:
```suggestion
                    return ReviewResponseDto.builder()
                            .reviewId(review.getId())
                            .starbucksId(starbucksReview.getStarbucks().getId())
                            .title(review.getTitle())
                            .content(review.getContent())
                            .build();
```

한 줄로 합쳐도 문제 없을 것 같아요!

Reply:
변수를 안 만들어도 됐을 것 같네요!! 한 줄로 수정했습니다~!!